gtdynamics/utils/FootContactState.h

```diff
@@ -0,0 +1,109 @@
+/* ----------------------------------------------------------------------------
+ * GTDynamics Copyright 2020, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * See LICENSE for the license information
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file  FootContactState.h
+ * @brief Utility methods for generating FootContactState objects.
+ * @author: Disha Das, Frank Dellaert
+ */
+
+#pragma once
+
+#include <gtdynamics/dynamics/DynamicsGraph.h>
+#include <gtdynamics/kinematics/Kinematics.h>
+#include <gtdynamics/universal_robot/Robot.h>
+#include <gtdynamics/utils/ConstraintSpec.h>
+
+#include <iosfwd>
+
+namespace gtdynamics {
+/**
+ * @class FootContactState class stores information about a robot stance
+ * and its duration.
+ */
+
+using ContactPointGoals = std::map<std::string, gtsam::Point3>;
+
+class FootContactState : public ConstraintSpec {
+ protected:
+  PointOnLinks contact_points_;  ///< Contact Points
+
+ public:
+  /// Constructor
+  FootContactState() {};
+
+  /**
+   * @fbrief Constructor with all contact points, takes list of PointOnLinks.
+   *
+   * @param[in] points_on_links List of link PointOnLinks.
+   */
+  FootContactState(const std::vector<PointOnLink> &points_on_links);
+
+  /**
+   * @fbrief Constructor with all contact points, takes a number of links and
+   * creates same contact points on all links.
+   *
+   * @param[in] links           List of link pointers.
+   * @param[in] contact_in_com  Point of contact on link.
+   */
+  FootContactState(const std::vector<LinkSharedPtr> &links,
+                   const gtsam::Point3 &contact_in_com);
+
+  /// Returns all the contact points in the stance
+  const PointOnLinks &contactPoints() const { return contact_points_; }
+
+  /// Check if phase has a contact for given link.
+  bool hasContact(const LinkSharedPtr &link) const {
+    int link_count =
+        std::count_if(contact_points_.begin(), contact_points_.end(),
+                      [&](const PointOnLink &contact_point) {
+                        return contact_point.link->name() == link->name();
+                      });
+    return link_count > 0;
+  }
+
+  // NOTE DISHA: Can modify this function to return multiple contact points on a
```

Comment:
kill note. Weird as I thought I had already done that in base branch

Reply:
done